docs/architecture/decisions/0005-IAsyncEnumerable-HttpResponseExceptionFilter.md

```diff
@@ -0,0 +1,20 @@
+# 5. WebAPI to use IAsyncEnumerable to stream potentially large responses.
+
+## Status
+Denied
+
+## Context
+There are currently two endpoints (`emissions/forecasts/batch`, `emissions/average-carbon-intensity/batch`) that require passing a payload as an array of items. This array can have as many items as the user wants, and it can take sometime to be processed which incurs in a delay to the client, which might feel that the service is irresponsive. Also, since these requests return enumerable items, and those are buffered before the client gets it, it impacts the overall memory footprint of the WebApp. Changing the signature of these endpoints to return an `IAsyncEnumerable` collection streams the response and helps to deal with these memory concerns.
```

Comment:
Nit:  'sometime' should be 'some time' and 'incurs in' should be 'incurs' (or even better would be the word 'creates').